src/quests/src/task.c

```diff
@@ -26,9 +27,16 @@ int mission_init(mission_t *mission, char *target_name, mission_types_t type)
     assert(mission != NULL);
     assert(target_name != NULL);
 
-    mission->target_name = target_name;
+    mission->target_name = strdup(target_name);
     mission->type = type;
 
+    if(type == MEET_NPC || type == KILL_NPC) {
+        // Make the string lowercase
+        for(char *p = mission->target_name; *p; p++) {
+            *p = *p > 0x40 && *p < 0x5b ? *p | 0x60 : *p;
+        }
+    }
```

Comment:
this seems to be some sort of copy and paste situation but a couple of things: make sure you're following the style guidelines (https://uchicago-cs.github.io/dev-guide/style_guide_c.html) (indentation and spaces being of concern), and to make the string lowercase I would just use the `case_insensitize()` function to reduce complexity and keep string manipulation consistent throughout chiventure.
```suggestion
    if (type == MEET_NPC || type == KILL_NPC) 
    {
        case_insensitize(mission->target_name);
    }
```
and honestly it might be better to just case_insensitize the `char *target_name` input from the get-go before even assigning `mission->target_name` to it

Reply:
Can we leave this here for now? It is fixed in a later branch already

Reply:
I mean i would just go ahead and fix it rq, you can just click on the suggestion to commit it if you'd like.

Reply:
There are more changes than just that need to go into fixing the actual issues with this function though. Changing this here will create a merge conflict for future branches that already fixed them all (see quests/wdl_file_branch for the fixes). LIke the entire other branch already has everything fixed and will get pulled in once this and an action management pr is done, so changing it here will only cause harm.

Reply:
if you're trying to avoid merge conflicts wouldn't you want this to be the same as dev?

Reply:
I can't make the other changes in this branch because they involve changing all mission names to lowercase and not just the ones provided which is going to crash a bunch of quest tests, which would be even more to fix in this 1700 line pr.

Reply:
Please I fixed everything else, just let this one thing remain, I promise it will be better in dev once everything else is sorted out with action management. We've had way too many problems with merge conflicts already and I don't want to cause any more pain with that.

Reply:
> We've had way too many problems with merge conflicts already and I don't want to cause any more pain with that.

I'm pretty sure this is going to become a merge conflict

Reply:
Fiiinneee, fixed but please approve now :)

Reply:
nah

Reply:
https://github.com/uchicago-cs/chiventure/commit/9f7a44eff2c6d547ca36015fcbbfc1e20ac57818
and also, respectfully, please grow up.